static-typing/analysis.md

```diff
@@ -0,0 +1,44 @@
+# Param - Static Typing - Analysis
+
+This document fixes the problem described in [problem.md](problem.md).
+
+## Updated Findings (2025)
+
+### Param Descriptors and Typing
+- Param descriptors (e.g., param.String, param.Number) can provide good attribute type inference in mypy if their stubs define correct __get__ return types (e.g., str, Optional[str], float).
+- Type annotations on descriptor assignments cause mypy errors unless suppressed with # type: ignore[assignment].
+- Wrapper functions using @overload (e.g., param_string) can help mypy infer the type of the value returned by the function, but the attribute is still a descriptor, so __init__ signatures are not synthesized.
```

Comment:
Why is the `__init__` signature not correct when the attribute is a selector?

Reply:
Sorry. Don't understand the question?

Reply:
Sorry I used the wrong word and meant descriptor instead of selector.

Why is the `__init__` signature not correct when the attribute is a descriptor?